### PR TITLE
Add more Authentication Strategies

### PR DESCRIPTION
**Pull request by hexa2k9**
Originally done in #32, this addresses PR Feedback.

### CODE REVIEW

**Comment by CSharpRU** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/AuthenticationStrategies/LdapAuthenticationStrategy.php  1
- src/AuthenticationStrategies/OktaAuthenticationStrategy.php  1
- src/AuthenticationStrategies/RadiusAuthenticationStrategy.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CSharpRU/vault-php/pullRequest?prid=5570266&bid=18265842)

**Comment by CSharpRU** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/AuthenticationStrategies/LdapAuthenticationStrategy.php  1
- src/AuthenticationStrategies/OktaAuthenticationStrategy.php  1
- src/AuthenticationStrategies/RadiusAuthenticationStrategy.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CSharpRU/vault-php/pullRequest?prid=5570266&bid=18265842)

**Comment by CSharpRU** [comment]
LGTM, just leave `extends` on the same line.

Also it would be great if you can add abstract base class for auths with path thing and just extend them from that base class. But for now you can leave it as it is, thanks!

**Comment by hexa2k9** [comment]
I've updated the `extends` and will work on an Abstract once I find a considerable amount of free time.

**Comment by CSharpRU** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/AuthenticationStrategies/LdapAuthenticationStrategy.php  1
- src/AuthenticationStrategies/OktaAuthenticationStrategy.php  1
- src/AuthenticationStrategies/RadiusAuthenticationStrategy.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CSharpRU/vault-php/pullRequest?prid=5570266&bid=18265842)

File: src/AuthenticationStrategies/AbstractUserPassAuthenticationStrategy.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public function authenticate(): Auth
+    {
+        if (strlen($this->methodPathSegment) >= 1) {
+            $response = $this->client->write(
+                sprintf('/auth/%s/login/%s', $this->methodPathSegment, $this->username),
+                ['password' => $this->password]
+            );
+
+            return $response->getAuth();
+        } else {
```
Comment (CSharpRU) [inline src/AuthenticationStrategies/AbstractUserPassAuthenticationStrategy.php:70]: Codacy Issue found: [The method authenticate uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/manual/CSharpRU/vault-php/pullRequest?prid=5570266)

**Comment by CSharpRU** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/AuthenticationStrategies/AbstractUserPassAuthenticationStrategy.php  2
- src/AuthenticationStrategies/LdapAuthenticationStrategy.php  1
- src/AuthenticationStrategies/RadiusAuthenticationStrategy.php  1
- src/AuthenticationStrategies/OktaAuthenticationStrategy.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CSharpRU/vault-php/pullRequest?prid=5570266&bid=18265842)

**Pull request merged**